pkg/leeway/build.go

```diff
@@ -1036,9 +1037,39 @@ func (p *Package) buildGo(buildctx *buildContext, wd, result string) (res *packa
 	}
 
 	var (
-		commands  [][]string
-		goCommand = "go"
+		commands      [][]string
+		isGoWorkspace bool
+		workFile      *modfile.WorkFile
 	)
+	if fc, err := os.ReadFile(filepath.Join(p.C.W.Origin, "go.work")); err == nil {
+		isGoWorkspace = true
+
+		workFile, err = modfile.ParseWork("go.work", fc, nil)
+		if err != nil {
+			return nil, xerrors.Errorf("cannot read go.work file: %w", err)
+		}
+		// we drop all use statements and later add the correct ones back in, starting with the src itself
```

Comment:
For understanding: I'm expecting we're removing the `use` statements so that we can generate a go.work for the package which only contains the "workspace" for the current package we're building. Based on that I was expecting to find somewhere later the code where all go package dependencies of the package would be traversed and `workFlie.AddNewUse` invoked - but I can't seem to find that in this diff. Did I overlook it or is my understanding of the approach off? ☺️

Reply:
Your expectation is correct :)
The code is here: https://github.com/gitpod-io/leeway/blob/73460fe17107a95d39c922d203ed5de18fbb14ab/pkg/leeway/build.go#L1101

I opted to use `go work use` instead of modifying the `go.work` file directly because this way it's more visible what happens. I.e. when you run leeway with `-v` you'll get a list of `go work use` commands printed. Originally I wanted to do all editing of the go work file this way (for the same reason), but it would have been to complicated to remove the old use statements, and we'd have required `jq` to be present in the userland.

Reply:
That makes a ton of sense, thanks for clarifying. When I read the "go work use" it had stowed it away as "Go should use the workspace file" 😝